src/scheduler/buckets.cpp

```diff
@@ -1135,10 +1135,10 @@ check_node_buckets(status *policy, server_info *sinfo, queue_info *qinfo, resour
 	} else if (!(sinfo->svr_to_psets.empty())) {
 		/* Find buckets associated with nodes of the server which owns the job */
 		for (auto &spset: sinfo->svr_to_psets) {
-			if (spset.svr_inst_id == resresv->svr_inst_id) {
+			if (spset.first == resresv->svr_inst_id) {
```

Comment:
The reason for using an unordered map was the O(1) lookups.  You just need to pass sinfo->svr_to_psets[resresv->svcr_inst_id].second->bkts to map_buckets()